CMakeLists.txt

```diff
@@ -94,7 +83,44 @@ else()
   set(CMAKE_EXE_LINKER_FLAGS_INIT "${CMAKE_EXE_LINKER_FLAGS_INIT} -Wl,-z,defs")
 endif()
 
-add_subdirectory(src)
+set(IRODS_ENABLE_GENQUERY2 OFF CACHE BOOL "Enable GenQuery2 support. GenQuery2 package must be installed.")
+
+set(CMAKE_IRODS_HTTP_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+set(CMAKE_IRODS_HTTP_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}")
+
+find_package(CURL REQUIRED)
+find_package(nlohmann_json "3.6.1" REQUIRED)
+find_package(OpenSSL REQUIRED COMPONENTS Crypto SSL)
+
+# jwt library to handle OIDC JWT responses containing user information
+find_package(jwt-cpp REQUIRED
+  HINTS "${IRODS_EXTERNALS_FULLPATH_JWTCPP}")
+
+include(ObjectTargetHelpers)
+
+add_subdirectory(core)
+add_subdirectory(endpoints)
+
+add_executable(${IRODS_HTTP_API_BINARY_NAME})
+target_link_objects(
+  ${IRODS_HTTP_API_BINARY_NAME}
+  PRIVATE
+  irods_http_api_core
+  irods_http_api_shared_operations
+  irods_http_api_endpoint_authentication
+  irods_http_api_endpoint_collections
+  #irods_http_api_endpoint_config
+  irods_http_api_endpoint_data_objects
+  irods_http_api_endpoint_information
+  irods_http_api_endpoint_query
+  irods_http_api_endpoint_resources
+  irods_http_api_endpoint_rules
+  irods_http_api_endpoint_tickets
+  irods_http_api_endpoint_users_groups
+  irods_http_api_endpoint_zones
```

Comment:
I did consider this and decided against it. Given how variable scoping in CMake works, this would just make the CMake really messy.

Reply:
I'm a little confused.

Wouldn't introducing variables be identical to how we handle `CMAKE_IRODS_HTTP_SOURCE_DIR` above or is there more to it that I'm not seeing?

Reply:
Oh, sorry, I think I misread/misinterpreted your original comment. I can do that.

Reply:
Before I go ahead with this, under what circumstances would we want to change the target name for an object library, but not the name of the variable containing the target name?

Reply:
I'm not sure. Perhaps the names will never change. I just noticed references across files and thought maybe we should use a variable so we can change things in one place.

This isn't a requirement either _(we're likely a long way from 1.0.0)_. Just trying to get everyone to think about it in case it hadn't crossed their mind.

Reply:
I don't think I'm going to make this change. I think it makes sense to have a variable name for the binary, but not for object libraries, as the variable names for object libraries would be tied too closely with the target names. If we change the target names, we'd probably end up changing the variable names too, defeating the purpose.